src/pages/ChattingRoom.tsx

```diff
@@ -0,0 +1,209 @@
+import SmallButton from "../components/SmallButton";
+import Profile from "../components/Profile";
+import LeftChat from "../components/LeftChat";
+import RightChat from "../components/RightChat";
+import { chatInterface } from "../interfaces/interface";
+import chatsData from "../json/chatsData.json";
+import Alert from "../components/Alert";
+
+const ChattingRoom = () => {
+  const currentUsers = useRecoilValue(currentUsersState);
+  const [typingUser, setTypingUser] = useRecoilState(typingUserState);
+  const nonTypingUser = currentUsers.filter(
+    (user) => user.userId !== typingUser.userId
+  )[0];
+  const [chatList, setChatList] = useState<chatInterface[]>(
+    chatsData.chattings[0].chatList
+  ); // TODO - recoil로 변경
+  // const [chatList, setChatList] = useRecoilState(currentChatListState);
+
+  const [inputText, setInputText] = useState<string>("");
+  const isTyping: boolean = inputText.trim() ? true : false;
+
+  const [isVisibleAlert, setIsVisibleAlert] = useState<boolean>(false);
+
+  const textareaRef = useRef<HTMLTextAreaElement>(null);
+  const bottomDivRef = useRef<HTMLDivElement>(null);
+
+  const handleInputChange = (e: React.ChangeEvent<HTMLTextAreaElement>) => {
+    setInputText(e.target.value);
+  };
+
+  const handleEnter = (e: React.KeyboardEvent<HTMLTextAreaElement>) => {
+    if (!e.shiftKey && e.key === "Enter") {
+      handleSubmit();
+    }
+  };
+
+  const handleSubmit = (e?: React.FormEvent<HTMLFormElement>) => {
+    e?.preventDefault();
+    setChatList([
+      ...chatList,
+      {
+        userId: typingUser.userId,
+        message: inputText,
+        date: String(new Date()),
+        chatId: `ex${chatList.length}`,
+      },
+    ]);
+    setInputText("");
+    textareaRef.current?.focus();
+  };
+
+  const hadleNotyetClick = () => {
+    setIsVisibleAlert(true);
+  };
+
+  useEffect(() => {
+    bottomDivRef.current?.scrollIntoView(false);
+  }, [chatList]);
+
+  useEffect(() => {
+    setTimeout(() => setIsVisibleAlert(false), 3000);
+  }, [isVisibleAlert]);
+
+  return (
+    <Wrapper>
+      <Header>
+        <SmallButton text={"<"} handleClick={hadleNotyetClick} />
+        <Profile nonTypingUser={nonTypingUser} setTypingUser={setTypingUser} />
+        <SmallButton text={"⋮"} handleClick={hadleNotyetClick} />
+      </Header>
+
+      <Main>
+        {chatList.map((chat: chatInterface) => {
+          if (chat.userId === typingUser.userId) {
+            return (
+              <RightChat
+                key={chat.chatId}
+                message={chat.message}
+                date={chat.date}
+              />
+            );
+          } else {
+            return (
+              <LeftChat
+                key={chat.chatId}
+                imgSrc={nonTypingUser.profileImage}
+                name={nonTypingUser.userName}
+                message={chat.message}
+                date={chat.date}
+              />
+            );
+          }
+        })}
```

Comment:
이럴 때에는 삼항 연산자를 쓰는 편이 가독성이 더 좋을 것 같네요!
```suggestion
        {chatList.map((chat: chatInterface) =>
          chat.userId === typingUser.userId ? (
            <RightChat
              key={chat.chatId}
              message={chat.message}
              date={chat.date}
            />
          ) : (
            <LeftChat
              key={chat.chatId}
              imgSrc={nonTypingUser.profileImage}
              name={nonTypingUser.userName}
              message={chat.message}
              date={chat.date}
            />
          )
        )}
```

사실 제가 생각하기에 더 좋은 방법은 `RightChat`과 `LeftChat` 컴포넌트를 따로 두지 않고, 하나의 컴포넌트 내에서 css를 달리 작성하여 중복되는 코드를 줄이고 컴포넌트의 재사용성을 높이면 좋을 것 같아요~

Reply:
삼항연산자가 더 깔끔할 것 같습니다!

말씀해주신 대로 이번 과제에서 하나의 컴포넌트로 사용하지 못한 점이 아쉬웠습니다.. 다음 과제에서는 props를 적극적으로 사용해 합쳐보고 싶습니다!